src/cookbook/effects/gradient-bubbles.md

```diff
@@ -148,35 +150,37 @@ a reference to the ancestor `ScrollableState`
 and a reference to the `BubbleBackground`’s
 `BuildContext`. Provide each of those to the `CustomPainter`.
 
-<!--skip-->
+<?code-excerpt "lib/bubble_painter.dart (ScrollableContext)" replace="/painter: //g"?>
 ```dart
-BubbleBackground(
-  scrollable: Scrollable.of(context)!,
+BubblePainter(
+  colors: colors,
   bubbleContext: context,
-  // ...
+  scrollable: ScrollableState(),
 ),
-
-//-----
-
+```
+<!-- this code excerpt adds an extra closing bracket
+at the end because the excerpt cuts off the paint method that's required for Custom Painter. -->
```

Comment:
Nice including the explanation for this :)

Reply:
With my memory, I was never going to be able to remember what that excerpt was doing even in 2 weeks ;)